pytket/qir/conversion/api.py

```diff
@@ -122,12 +133,42 @@ def pytket_to_qir(
             qir_int_type=int_type,
             trunc=trunc,
         )
+    elif profile == QIRProfile.AZUREADAPTIVE:
+        qir_generator = AzureAdaptiveProfileQirGenerator(
+            circuit=circ,
+            module=m,
+            wasm_int_type=int_type,
+            qir_int_type=int_type,
+            trunc=trunc,
+        )
     else:
         raise NotImplementedError("unexpected profile")
 
     populated_module = qir_generator.circuit_to_module(qir_generator.circuit, True)
 
-    if qir_generator.has_wasm:
+    if profile == QIRProfile.AZUREADAPTIVE:
+
+        assert not qir_generator.has_wasm
```

Comment:
My understanding is that we don't want to do wasm for azure.

Reply:
I think that's a fair assumption for now at least.